tezos-sig-gating-contracts/scripts/tests/gatednftminterSimpleMultisig.spec.ts

```diff
@@ -0,0 +1,358 @@
+    );
+    const currentStorage: any = await cntrSignerManager.storage();
+    lastProposalId = await currentStorage.next_proposal_id;
+    const op = await cntrSignerManager.methodsObject
+      .createAddOwnerProposal(frank)
+      .send();
+    // const op = await cntrSignerManager.methodsObject.addOwner(frank).send();
+    await op.confirmation(2);
+    // Verify
+    const storageAfter: any = await cntrSignerManager.storage();
+    const prop = await storageAfter.proposals.get(lastProposalId);
+    expect(prop.action?.addOwner === frank).to.be.true;
+
+    //Tezos.setSignerProvider(await InMemorySigner.fromSecretKey(frank_pk));
+    const op_validate = await cntrSignerManager.methodsObject
+      .validateProposal([lastProposalId, true])
+      .send();
+    await op_validate.confirmation(2);
+    // Verify
+    const storage: any = await cntrSignerManager.storage();
+    const frankAuth = await storage.owners.get(frank);
+    expect(frankAuth).to.be.true;
+  });
+
+  it(`Should setThreshold to 2 in multisig`, async () => {
+    const cntrSignerManager = await Tezos.contract.at(
+      exampleGatedNFTMinterCntrl ? exampleGatedNFTMinterCntrl : ""
+    );
+    const currentStorage: any = await cntrSignerManager.storage();
+    lastProposalId = await currentStorage.next_proposal_id;
+    const newThreshold = 2;
+    const op = await cntrSignerManager.methodsObject
+      .createSetThresholdProposal(newThreshold)
+      .send();
+    // const op = await cntrSignerManager.methodsObject.addOwner(frank).send();
+    await op.confirmation(2);
+
+    // Verify
+    const storageAfter: any = await cntrSignerManager.storage();
+    const prop = await storageAfter.proposals.get(lastProposalId);
+    expect(prop.action?.setThreshold.toNumber()).to.be.eq(newThreshold);
+
+    const op_validate = await cntrSignerManager.methodsObject
+      .validateProposal([lastProposalId, true])
+      .send();
+    await op_validate.confirmation(2);
+    // Verify
+    const storage: any = await cntrSignerManager.storage();
+    const threshold = await storage.threshold;
+    expect(threshold.toNumber() === newThreshold).to.be.true;
+  });
+
+  it(`Should create a proposal to ChangeSigner in multisig`, async () => {
+    const cntrSignerManager = await Tezos.contract.at(
+      exampleGatedNFTMinterCntrl ? exampleGatedNFTMinterCntrl : ""
+    );
+    const currentStorage: any = await cntrSignerManager.storage();
+    lastProposalId = await currentStorage.next_proposal_id;
+
+    const op = await cntrSignerManager.methodsObject
+      .createNewSignerProposal(alice)
+      .send();
+    await op.confirmation(2);
+    // Verify
+    const storage: any = await cntrSignerManager.storage();
+    const prop = await storage.proposals.get(lastProposalId);
+    expect(prop.action?.changeSigner === alice).to.be.true;
+    expect(storage.pause === false).to.be.true;
+  });
+
+  it(`Should validate proposal (Alice)`, async () => {
```

Comment:
I'm surprised this test works... the test should be written assuming parallel execution and not assumed a change of state from a previous test

Reply:
In order to write tests meant to be executed in parallel without having a lot of duplicate code, you should reactor each step in functions (add owner, changeThreshold, etc)